### PR TITLE
Image manager adding max file limit and alternative column structure

### PR DESCRIPTION
**Pull request by kvisca**
- Adding user-defined maximum file limit.
- Adding a 2 column layout needed in the sidebar
- Adding responsive for 4 column layout.

**2 Column:**
Screen Shot 2020-12-08 at 5 35 43 PM

**3 file max:**
Screen Shot 2020-12-08 at 5 36 15 PM

### CODE REVIEW

**Comment by robbieaverill** [comment]
What happens with max = 1?

**Comment by kvisca** [comment]
> What happens with max = 1?

You'll be able to have 1 file, which is what we want for the editor. 

Screen Shot 2020-12-08 at 5 46 37 PM

**Comment by robbieaverill** [comment]
Yum

**Review by robbieaverill — COMMENTED** [review]
Is there a difference between columns and maxFiles? Maybe maxFiles needs to be defined as a prop?

**Comment by kvisca** [comment]
> Is there a difference between columns and maxFiles? Maybe maxFiles needs to be defined as a prop?

Columns you can have either 2 or 4 column layouts. 

`maxFiles` is a prop that's completely separate from the columns. I didn't add it as a knob because it was acting very oddly if dynamically changed, assuming that's what you're referring to?

**Comment by robbieaverill** [comment]
No, it needs to be defined as a prop on the component that uses it

File: src/components/ChecImageManager.vue [hunk 1]
```diff
@@ -24,11 +24,15 @@
           @remove="() => removeFile(file)"
         />
       </Draggable>
-      <ChecButton @click="openDialog">
-        <ChecIcon class="chec-image-manager__icon" icon="image" size="base" /> {{ $t('imageManager.chooseImages') }}
-      </ChecButton>
-      <div v-if="footnote" class="chec-image-manager__helper">
-        {{ footnote }}
+      <div v-if="maxFiles === null || maxFiles > allFiles.length">
```
Comment (ScopeyNZ) [inline src/components/ChecImageManager.vue:27]:
```suggestion
      <template v-if="maxFiles === null || maxFiles > allFiles.length">
```

BYO closing tag. Makes more sense not to add the block if it isn't needed?

File: src/components/ChecImageManager.vue [hunk 2]
```diff
@@ -52,16 +56,52 @@ export default {
...
       default: null,
     },
+    /**
+     * Number of columns to be displayed. either 2 or 4.
+     */
+    columns: {
+      type: Number,
+      default: 4,
+      validator(value) {
+        return [2, 4].includes(value);
```
Comment (ScopeyNZ) [inline src/components/ChecImageManager.vue:73]: Did you need this validator? Seems like an unnecessary restriction...

**Review by robbieaverill — APPROVED** [review]

File: src/components/ChecImageManager.vue (hunk 1 above)
Comment (kvisca) [reply src/components/ChecImageManager.vue]: Good call, I've switched it out.

File: src/components/ChecImageManager.vue (hunk 2 above)
Comment (kvisca) [reply src/components/ChecImageManager.vue]: No, I suppose not, I've removed it.

**Pull request merged**